components/Action/ConfirmOnClick/index.js

```diff
@@ -0,0 +1,146 @@
+import PropTypes from 'prop-types';
+import React, { Fragment } from 'react';
+
+import Confirm from '../../View/Modal/Confirm';
+import TextInput from '../../Input/TextInput';
+import { isFalsy } from '../../../utils/common';
+
+import styles from './styles.scss';
+
+const propTypes = {
+    onClick: PropTypes.func,
+    disabled: PropTypes.bool,
+    challengeValue: PropTypes.string,
+    challengeLabel: PropTypes.string,
+    challengePlaceholder: PropTypes.string,
+    confirmationMessage: PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.node,
+    ]).isRequired,
+    confirmationTitle: PropTypes.string,
+};
+
+const defaultProps = {
+    onClick: () => {},
+    disabled: false,
+    confirmationTitle: undefined,
+    challengeValue: undefined,
+    challengeLabel: '',
+    challengePlaceholder: '',
+};
+
+const ConfirmOnClick = (WrappedComponent) => {
+    const ConfirmComponent = class extends React.PureComponent {
+        static propTypes = propTypes;
+        static defaultProps = defaultProps;
+
+        constructor(props) {
+            super(props);
+
+            this.state = {
+                showConfirm: false,
+                challengeTextInputValue: '',
+                challengeSuccess: false,
```

Comment:
No need to maintain separate state for challengeSuccess

Reply:
Fixed!